src/parse_instruction.py

```diff
@@ -200,15 +204,24 @@ def __str__(self) -> str:
 
 @dataclass(frozen=True)
 class InstructionMeta:
+    # True if the original asm line was marked with a goto pattern
     emit_goto: bool
+    # Asm source filename & line number
     filename: str
     lineno: int
+    # True if the Instruction is not directly from the source asm
     synthetic: bool
+    # True if the Instruction was part of a matched IR pattern, but not elided
+    in_pattern: bool
```

Comment:
I'm a bit hesistant to put this in InstructionMeta, which is right now kinda best-effort for debugging purposes. Would it work to put the bit in Instruction itself?